specification/communication/data-plane/role/preview/2022-01-01-preview/CommunicationRole.json

```diff
@@ -0,0 +1,439 @@
+        "tags": [
+          "Role"
+        ],
+        "summary": "Get a communication role.",
+        "operationId": "CommunicationRole_Get",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "id",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationErrorResponse"
+            }
+          },
+          "200": {
+            "description": "OK - Returns the communication role.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationRoleResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get a Communication Role": {
+            "$ref": "./examples/GetRole.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CommunicationRoleCreatePermissions": {
+      "description": "The object containing the various types of permissions.",
+      "required": [
+        "dataActions"
+      ],
+      "type": "object",
+      "properties": {
+        "dataActions": {
+          "description": "The list of data oriented action permissions.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CommunicationRolePermissionActions"
+          }
+        }
+      }
+    },
+    "CommunicationRoleCreateRequest": {
+      "required": [
+        "displayName",
+        "isEnabled",
+        "permissions"
+      ],
+      "type": "object",
+      "properties": {
+        "displayName": {
+          "description": "The human readable name of the Role.",
+          "type": "string"
+        },
+        "permissions": {
+          "$ref": "#/definitions/CommunicationRoleCreatePermissions"
+        },
+        "isEnabled": {
+          "description": "Is Role enabled for assignments.",
+          "type": "boolean"
+        }
+      }
+    },
+    "CommunicationRoleGetAllResponse": {
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The query response content",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CommunicationRoleResponse"
+          }
+        },
+        "maxItems": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "@nextLink": {
+          "type": "string"
+        }
+      }
+    },
+    "CommunicationRolePermissionActions": {
+      "description": "The available action permissions list.",
+      "enum": [
+        "communication.services/calling/audio/receive/action",
+        "communication.services/calling/video/receive/action",
+        "muteThemselves",
```

Comment:
This doesn't follow the pattern of the other actions, can we change that?

Reply:
Yes. It is a tmp name but I'll write it in a way it fits with the patteern.

Reply:
This is a temporary place holder. I don't have any proposed name for that yet.